FabLabKasse/kassenbuch.py

```diff
@@ -1277,19 +1277,19 @@ def fetch_input(explanation, default_input=None,
                 :type allowed_regexp: basestr | None
                 :type extra_checks: function | None
                 """
-                default_str = " [{0}]".format(unicode(default_input)) if default_input is not None else ""
+                default_str = u" [{0}]".format(unicode(default_input)) if default_input is not None else ""
                 allowed_regexp = allowed_regexp if allowed_regexp is not None else ur".*"
                 extra_checks = extra_checks if extra_checks else lambda x: True
 
                 while True:
                     input_str = raw_input(u"{e}{d}: ".format(
-                        e=explanation, d=default_str))
+                        e=explanation, d=default_str)).decode(sys.stdin.encoding)
 
                     if input_str == '' and default_input is not None:
```

Comment:
I am not sure whether comparison with an empty string is the same with and without unicode.  (is `u""` == `""`)

OK, i just checked it: both in Python 2 and 3 `'' == u''` evaluates to `True`